examples/nextjs/app-router/app/layout.tsx

```diff
@@ -0,0 +1,44 @@
+import '../styles/globals.css'
+import * as React from 'react'
+import {
+    DevCycleClientProvider,
+    getVariableValue,
+    identifyUser,
+    setSDKKey,
+} from '@devcycle/next-sdk/server'
+
+const serverIds = [
+    'server-1',
+    'server-2',
+    'server-3',
+    'server-4',
+    'server-5',
+    'server-6',
+    'server-7',
+    'server-8',
+    'server-9',
+    'server-10',
+]
+
+export default async function RootLayout({
+    // Layouts must accept a children prop.
+    // This will be populated with nested layouts or pages
+    children,
+}: {
+    children: React.ReactNode
+}) {
+    const randomId = serverIds[Math.floor(Math.random() * serverIds.length)]
+    return (
+        <html lang="en">
+            <body>
+                <DevCycleClientProvider
```

Comment:
this is a server component which handles initializing and fetching config etc. on serverside, making variables available to server components

It also renders a clientside provider with a bootstrapped client that uses the server's config data, guaranteeing hydration consistency. That bootstrapped client is also used on the server for rendering the first pass of client components